#2303 Mention NULL behavior in WHERE clause in docs (#2379)
slick/slick: doc/paradox/sql-to-slick.md

@@snip [SqlToSlick.scala](../code/SqlToSlick.scala) { #sqlQueryFilter }

<h4>Slick</h4>

> Scala's equivalent for `WHERE` is `filter`. There are also the convenience variations `filterIf` and `filterOpt`.

Reviewer: ```suggestion
```suggestion
Scala's equivalent for `WHERE` is
@scaladoc[`filter`](scaladoc:slick.lifted.Queryl#filter[T](f:E=>T)(implicitwt:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C]).
There are also the convenience variations
@scaladoc[`filterIf`](slick.lifted.Query#filterIf[T](p:Boolean)(f:E=%3ET)(implicitevidence$2:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C])
and
@scaladoc[`filterOpt`](slick.lifted.Query#filterOpt[V,T](optValue:Option[V])(f:(E,V)=>T)(implicitevidence$1:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C]).
```

Reviewer: I hope the links work

Author: Incorporated. Is it possible to check the built version?

Reviewer: If you want to check locally just run `sbt paradox`. Otherwise we can just merge it.

Reviewer: Or `sbt "show paradox"`, then it will print where the generated docs are

Reviewer: Or just run scaladocCheckLinks. I should add that to CI.

Author: I just tried this out, got couple of errors, I think due to the encoding of `=>`, fixed them as `=>%3E` and it worked. Though I think scaladoc needs a little work to make the selected method a little more prominent :D I can see that its description is expanded.

> Scala's equivalent for `WHERE` is `filter`. There are also the convenience variations `filterIf` and `filterOpt`.
> 
> @@@ note
> 
> In Slick, `Option` types correspond to nullability in SQL. If a filter predicate has type `Rep[Option[Boolean]]` that means the database can evaluate it to `TRUE`, `FALSE`, or `NULL`, but `NULL` behaves like `FALSE`. This behavior is consistent across `filter`, `filterIf`, `filterOpt`, and `withFilter`.
> 
> @@@
> 
> @@@ warning
> 
> Make sure to use `===` instead of `==` for comparison, so that Slick can convert the expression to an SQL `=`. Slick uses `===` for that because `==` is already used by Scala for performing comparisons locally.
> 
> @@@

Reviewer: ```suggestion
Scala's equivalent for `WHERE` is
@scaladoc[`filter`](scaladoc:slick.lifted.Queryl#filter[T](f:E=>T)(implicitwt:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C]).
There are also the convenience variations
@scaladoc[`filterIf`](slick.lifted.Query#filterIf[T](p:Boolean)(f:E=%3ET)(implicitevidence$2:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C])
and
@scaladoc[`filterOpt`](slick.lifted.Query#filterOpt[V,T](optValue:Option[V])(f:(E,V)=>T)(implicitevidence$1:slick.lifted.CanBeQueryCondition[T]):slick.lifted.Query[E,U,C]).
@@@ note
In Slick, `Option` types correspond to nullability in SQL. If a filter predicate has type `Rep[Option[Boolean]]` that means the database can evaluate it to `TRUE`, `FALSE`, or `NULL`, but `NULL` behaves like `FALSE`. This behavior is consistent across `filter`, `filterIf`, `filterOpt`, and `withFilter`.
@@@
@@@ warning
Make sure to use `===` instead of `==` for comparison, so that Slick can convert the expression to an SQL `=`. Slick uses `===` for that because `==` is already used by Scala for performing comparisons locally.
@@@
```

@@snip [SqlToSlick.scala](../code/SqlToSlick.scala) { #sqlQueryFilter }

<h4>Slick</h4>

> Scala's equivalent for `WHERE` is `filter`. Make sure to use `===` instead of `==` for comparison. Slick also preserves the SQL behavior throughout filter-like methods (`filter`, `filterIf`, `filterOpt`, `withFilter`), where the expressions involving `NULL` (corresponding to `None` in Scala/Slick) yield `NULL`, hence evaluated as `false` and such rows are filtered out. 

Reviewer: ```suggestion
Scala's equivalent for `WHERE` is `filter`. There are also the convenience variations `filterIf` and `filterOpt`.

@@@ note

In Slick, `Option` types correspond to nullability in SQL. If a filter predicate has type `Rep[Option[Boolean]]` that means the database can evaluate it to `TRUE`, `FALSE`, or `NULL`, but `NULL` behaves like `FALSE`. This behavior is consistent across `filter`, `filterIf`, `filterOpt`, and `withFilter`.

@@@

@@@ warning

Make sure to use `===` instead of `==` for comparison, so that Slick can convert the expression to an SQL `=`. Slick uses `===` for that because `==` is already used by Scala for performing comparisons locally.

@@@
```

Author: Much better, thank you.
